src/Psl/Iter/times.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Psl\Iter;
+
+use Psl;
+
+/**
+ * Creates an iterator containing all numbers between the start and end value
+ * (inclusive) with a certain step.
+ *
+ * Examples:
+ *
+ *     Iter\times(5, Fun\Identity())
+ *     => Iter(1, 2, 3, 4, 5)
+ *
+ * @psalm-template T
+ *
+ * @psalm-param int $number How many times should the factory be executed?
+ * @psalm-param (callable(int): T) $factory
+ *
+ * @psalm-return Iterator<int, T>
+ *
+ * @throws Psl\Exception\InvariantViolationException If $number < 1
+ */
+function times(int $number, callable $factory): Iterator
+{
+    Psl\invariant($number >= 1, 'The number of times you want to iterate must be at least 1.');
```

Comment:
```suggestion
    Psl\invariant(null === $multiplier || $multiplier >= 0, 'Expected a non-negative multiplier');
```

Reply:
It isn't really multiplying but incrementing ... Or do you want me to make $numer multiplyable? Because that sounds more like a case for `map()`.

Reply:
I just copy-pasted this from Iter\repeat 😅 we can keep it `$number`